### PR TITLE
feat(cli-repl): add tlsCertificateSelector support on Win32 MONGOSH-573

### PR DESCRIPTION
**Pull request by addaleax**
No tests so far (including no manual ones), and I think testing this automatically would have to involve stubbing out the `win-export-certificate-and-key` package.

### CODE REVIEW

File: packages/cli-repl/test/e2e-tls.spec.ts [hunk 1]
```diff
@@ -181,7 +186,8 @@ describe('e2e TLS', () => {
       );
       const certUser = 'emailAddress=tester@example.com,CN=Wonderwoman,OU=DevTools Testers,O=MongoDB';
 
-      it('can connect with cert to create user', async() => {
+      before(async() => {
+        /* connect with cert to create user */
```
Comment (addaleax) [inline packages/cli-repl/test/e2e-tls.spec.ts:190]: ... took me half an hour to figure out why the server couldn't find the user :)

**Review by addaleax — COMMENTED** [review]
This should be ready for review -- I've manually verified that this works with actual certificate/private key combinations from the Windows CA store.

File: packages/cli-repl/test/e2e-tls.spec.ts (hunk 1 above)
Comment (rose-m) [reply packages/cli-repl/test/e2e-tls.spec.ts:190]: Sorrrryyy :D

File: packages/cli-repl/src/arg-parser.ts [hunk 2]
```diff
@@ -106,9 +106,7 @@ const DEPRECATED_ARGS_WITH_REPLACEMENT: Record<string, string> = {
 const UNSUPPORTED_ARGS: Readonly<string[]> = [
   'eval',
   'sslFIPSMode',
-  'tlsFIPSMode',
-  'sslCertificateSelector',
```
Comment (rose-m) [inline packages/cli-repl/src/arg-parser.ts:110]: We should probably update the help text of the CLI to include the flags and also add a comment in there to say that it's currently only supported on Windows.

**Review by rose-m — APPROVED** [review]
Awesome :) I just think we should update the usage / help output and the README to document our newly supported flag :D

**Comment by addaleax** [comment]
> Awesome :) I just think we should update the usage / help output and the README to document our newly supported flag :D

@rose-m I’ve updated the README/help output, but the flag was already documented – it should not have been, right?

**Comment by rose-m** [comment]
Yeah it should have been removed by my other PR where I cleaned those up... anyway - now looks good ✅

**Pull request merged**